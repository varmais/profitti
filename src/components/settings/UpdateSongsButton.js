import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {
  Button,
  PixelRatio,
  StyleSheet,
  Text,
  View
} from 'react-native';
import { connect } from 'react-redux';
import Spinner from 'react-native-loading-spinner-overlay';
import config from '../../config';
import { fetchSongs } from '../../redux/songs';

const styles = StyleSheet.create({
  container: {
    height: 60,
    backgroundColor: config.colors.white,
    borderWidth: 0,
    justifyContent: 'center'
  },
  text: {
    color: config.colors.blackLight,
    fontSize: 18
  },
  button: {
    position: 'absolute',
    right: 0,
    bottom: 27 / PixelRatio.get(),
    width: 70
  }
});

export class UpdateSongsButton extends Component {
  static propTypes = {
    loading: PropTypes.bool.isRequired,
    fetchSongs: PropTypes.func.isRequired
  };

  render () {
    const { loading, fetchSongs } = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.text}>Päivitä laulut</Text>
        <View style={styles.button}>
          <Button
            disabled={loading}
            onPress={fetchSongs}
            title="Päivitä"
            color={config.colors.green}
            accessibilityLabel="Päivitä laulut"
          />
        </View>
        <Spinner visible={loading} />
      </View>
    );
  }
}

export default connect((state) => ({
  loading: state.songs.loading
}), {
  fetchSongs
})(UpdateSongsButton);