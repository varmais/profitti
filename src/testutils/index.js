import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

global.React = React;
global.renderer = renderer;

export function createNavigationProp (params = {}) {
  return {
    state: {params: params},
    navigate: () => {}
  }
}