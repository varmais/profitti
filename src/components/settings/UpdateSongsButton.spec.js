import '../../testutils';
import { UpdateSongsButton } from './UpdateSongsButton';

describe('components:settings:UpdateSongsButton', () => {
  let fetchSongsStub;

  beforeEach(() => {
    fetchSongsStub = jest.fn();
  });

  describe('when songs are not loading', () => {
    test('renders UpdateSongsButton', () => {
      const snapshot = renderer.create(<UpdateSongsButton loading={false} fetchSongs={fetchSongsStub} />).toJSON();
      expect(snapshot).toMatchSnapshot();
    });

    describe('when button is pressed', () => {
      test('it calls fetchSongs callback');
    });
  });

  describe('when songs are loading', () => {
    test('renders UpdateSongsButton', () => {
      const snapshot = renderer.create(<UpdateSongsButton loading={true} fetchSongs={fetchSongsStub} />).toJSON();
      expect(snapshot).toMatchSnapshot();
    });
  });
});