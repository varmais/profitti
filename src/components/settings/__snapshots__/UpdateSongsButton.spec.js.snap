// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:settings:UpdateSongsButton when songs are loading renders UpdateSongsButton 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderWidth": 0,
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#222",
        "fontSize": 18,
      }
    }
  >
    Päivitä laulut
  </Text>
  <View
    style={
      Object {
        "bottom": 13.5,
        "position": "absolute",
        "right": 0,
        "width": 70,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Päivitä laulut"
      accessibilityTraits={
        Array [
          "button",
          "disabled",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#9EB741",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Päivitä
        </Text>
      </View>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flex": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "bottom": 0,
              "flex": 1,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "height": 50,
                  "top": 80,
                },
                undefined,
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`components:settings:UpdateSongsButton when songs are not loading renders UpdateSongsButton 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderWidth": 0,
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#222",
        "fontSize": 18,
      }
    }
  >
    Päivitä laulut
  </Text>
  <View
    style={
      Object {
        "bottom": 13.5,
        "position": "absolute",
        "right": 0,
        "width": 70,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Päivitä laulut"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#9EB741",
              },
            ]
          }
        >
          Päivitä
        </Text>
      </View>
    </View>
  </View>
  <View />
</View>
`;
